plugins/webhooks/server/api/webhookSubscriptions.ts

```diff
@@ -120,37 +98,24 @@ router.post(
   validate(T.WebhookSubscriptionsUpdateSchema),
   transaction(),
   async (ctx: APIContext<T.WebhookSubscriptionsUpdateReq>) => {
-    const { id, name, url, secret } = ctx.input.body;
+    const { id, name, url, secret, events } = ctx.input.body;
     const { user } = ctx.state.auth;
     const { transaction } = ctx.state;
-    const events: string[] = compact(ctx.input.body.events);
-    const webhookSubscription = await WebhookSubscription.findByPk(id, {
+
+    let webhookSubscription = await WebhookSubscription.findByPk(id, {
       rejectOnEmpty: true,
       lock: transaction.LOCK.UPDATE,
       transaction,
     });
 
     authorize(user, "update", webhookSubscription);
 
-    await webhookSubscription.update(
-      {
-        name,
-        url,
-        events,
-        enabled: true,
-        secret: isEmpty(secret) ? undefined : secret,
-      },
-      { transaction }
-    );
-
-    await Event.createFromContext(ctx, {
-      name: "webhookSubscriptions.update",
-      modelId: webhookSubscription.id,
-      data: {
-        name: webhookSubscription.name,
-        url: webhookSubscription.url,
-        events: webhookSubscription.events,
-      },
+    webhookSubscription = await webhookSubscription.updateWithCtx(ctx, {
```

Comment:
Is the reassignment needed? If so might need to change `updateWithCtx` to match `update` behavior

Reply:
It ends up being an unnecessary assignment - changed it now.